pkg/commands/componentarchive/remote/remote_test.go

```diff
@@ -138,12 +139,101 @@ var _ = Describe("Remote", func() {
+		expectedRef := srcRepoCtxURL + "/component-descriptors/" + cd.Name
+		Expect(err).ToNot(HaveOccurred())
+		Expect(repos).To(ContainElement(Equal(expectedRef)))
+
+		manifest, err := client.GetManifest(ctx, srcOCIRef)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(manifest.Layers).To(HaveLen(2))
+		Expect(manifest.Layers[0].MediaType).To(Equal(cdoci.ComponentDescriptorTarMimeType),
+			"Expect that the first layer contains the component descriptor")
+
+		compResolver := cdoci.NewResolver(client)
+		sourceComp, err := compResolver.Resolve(ctx, cdv2.NewOCIRegistryRepository(srcRepoCtxURL, ""), cd.Name, cd.Version)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(sourceComp.Name).To(Equal(cd.Name))
+		Expect(sourceComp.Resources[len(sourceComp.Resources)-1].Access.Type).To(Equal("localOciBlob"), "Expect that the localFilesystem blob has been correctly converted to a localOciBlob")
+
+		var layerBlob bytes.Buffer
+		Expect(client.Fetch(ctx, srcOCIRef, manifest.Layers[1], &layerBlob)).To(Succeed(), "Expect that the second layer contains the local blob")
+		Expect(layerBlob.String()).To(Equal(blobContent))
+
+		copyOpts := &remote.CopyOptions{
+			OciOptions: options.Options{
+				AllowPlainHttp:     false,
+				RegistryConfigPath: "/auth.json",
+			},
+		}
+		copyOpts.SourceRepository = srcRepoCtxURL
+		copyOpts.ComponentName = cd.Name
+		copyOpts.ComponentVersion = cd.Version
+		copyOpts.TargetRepository = targetRepoCtxURL
+
+		Expect(copyOpts.Run(ctx, logr.Discard(), testdataFs)).To(Succeed())
+
+		repos, err = client.ListRepositories(ctx, targetRepoCtxURL)
+		Expect(err).ToNot(HaveOccurred())
+
+		targetOCIRef, err := components.OCIRef(cdv2.NewOCIRegistryRepository(targetRepoCtxURL, ""), cd.Name, cd.Version)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(repos).To(ContainElement(Equal(targetRepoCtxURL+"/component-descriptors/"+cd.Name)), "Expect that the repositories contains target repo")
+
+		manifestTarget, err := client.GetManifest(ctx, targetOCIRef)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(manifestTarget.Layers).To(HaveLen(2))
+		Expect(manifestTarget.Layers[0].MediaType).To(Equal(cdoci.ComponentDescriptorTarMimeType),
+			"Expect that the first layer contains the component descriptor")
+		Expect(manifestTarget.Layers[1].MediaType).To(Equal("text/plain"),
+			"Expect that the second layer contains the local blob")
+
+		targetComp, err := compResolver.Resolve(ctx, cdv2.NewOCIRegistryRepository(targetRepoCtxURL, ""), cd.Name, cd.Version)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(targetComp.Name).To(Equal(cd.Name))
+		Expect(targetComp.Resources[len(targetComp.Resources)-1].Access.Type).To(Equal("localOciBlob"), "Expect that the localFilesystem blob has been correctly converted to a localOciBlob")
+
+		var layerBlobTarget bytes.Buffer
+		Expect(client.Fetch(ctx, targetOCIRef, manifest.Layers[1], &layerBlobTarget)).To(Succeed())
```

Comment:
Hi, 
I would have used `manifestTarget` instead of `manifest` to fetch the target blob.
Besides this, all looks good for me.

Reply:
thank you for the hint, committed the fix